# HG changeset patch
# User Mike Miller <mtmiller@octave.org>
# Date 1505764202 25200
#      Mon Sep 18 12:50:02 2017 -0700
# Node ID 6782105f18206ac96d0cff2b5b68d9460abe0559
# Parent  ba164db1c4a4b121d722dd769dd291d96b018182
imfinfo.m: Fix error when a second argument is used (bug #52041).

* imfinfo.m: Use "varargin{1}" rather than "ext" as function prototype was
changed.

diff --git a/scripts/image/imfinfo.m b/scripts/image/imfinfo.m
--- a/scripts/image/imfinfo.m
+++ b/scripts/image/imfinfo.m
@@ -149,17 +149,17 @@
 ## Author: Soren Hauberg <hauberg@gmail.com>
 
 function info = imfinfo (filename, varargin)
 
   if (nargin < 1 || nargin > 2)
     print_usage ();
   elseif (! ischar (filename))
     error ("imfinfo: FILENAME must be a string");
-  elseif (nargin > 1 && ! ischar (ext))
+  elseif (nargin > 1 && ! ischar (varargin{1}))
     error ("imfinfo: EXT must be a string");
   endif
 
   info = imageIO ("imfinfo", @__imfinfo__, "info", filename, varargin{:});
 
 endfunction
 
 
@@ -176,8 +176,14 @@ endfunction
 %!     imread ("I_sure_hope_this_file_does_not_exist.jpg");
 %!   catch
 %!     error_thrown = true;
 %!   end_try_catch
 %! unwind_protect_cleanup
 %!   imformats ("update", "jpg", fmt_ori);
 %! end_unwind_protect
 %! assert (error_thrown, true);
+
+## Test input validation
+%!error imfinfo ()
+%!error imfinfo (1,2,3)
+%!error <FILENAME must be a string> imfinfo (1)
+%!error <EXT must be a string> imfinfo ("foo", 1)
